dask/order.py

```diff
@@ -326,6 +326,8 @@ def process_runnables() -> None:
                             if num_needed[current] <= 1:
                                 for k in path:
                                     add_to_result(k)
+                            else:
+                                runnable_hull.discard(current)
```

Comment:
the loop invariant this should preserve is that runnable_hull and known_runnable_paths are disjoint

Reply:
ok, I stand corrected... that invariant is still not true. I'll investigate...